utils/gen_onnx_mlir.py

```diff
@@ -399,6 +399,11 @@
     'Sub',
     'Xor',
 ]
+broadcast_to_bool_list = [
```

Comment:
This list and `custom_builder_broadcast_ops_list` are overlapped, which is confusing to users.

Could we do the same as Line 408 for `custom_builder_ops_list`, e.g.
- rename `custom_builder_broadcast_ops_list` to `custom_builder_broadcast_to_same_type_ops_list`
- rename `broadcast_to_bool_list` to `custom_builder_broadcast_to_bool_ops_list`
- then union them: `custom_builder_broadcast_ops_list = custom_builder_broadcast_to_same_type_ops_list +  custom_builder_broadcast_to_bool_ops_list`

Reply:
Yeah sure! That makes sense. I'll get right on that